src/kiara_plugin/topic_modelling/modules/corpus_preparation.py

```diff
@@ -30,83 +25,221 @@ def create_inputs_schema(self):
+        
+        if table_obj.is_set:
 
-        except Exception as e:
-            pass;
+            sources: Union[pa.Table, None] = table_obj.data
+            
+            assert sources is not None
 
+            sources_col_names = sources.column_names
+
+            if column_name not in sources_col_names:
+
+                raise KiaraProcessingException(
+                    f"Could not find file names column '{column_name}' in the table. Please specify a valid column name manually, using one of: {', '.join(sources_col_names)}"
+                )
+            
+            sources_data: pa.Table = table_obj.data.arrow_table
+            
+            sources_tb: pl.DataFrame = pl.from_arrow(sources_data) # type: ignore
 
-        sources = pl.from_arrow(table_obj.data.arrow_table)
 
         def get_ref(file):
             try:
-                ref_match = re.findall(r'(\w+\d+)_\d{4}-\d{2}-\d{2}_',file)
+                ref_match = re.findall(r"(\w+\d+)_\d{4}-\d{2}-\d{2}_", file)
                 if not ref_match:
                     return None
                 return ref_match[0]
-            
+
             except Exception as e:
-                print(f"Error in get_ref: {e}")
-                return None
+                raise KiaraProcessingException(e)
 
         def get_date(file):
             try:
-                date_match = re.findall(r'_(\d{4}-\d{2}-\d{2})_',file)
+                date_match = re.findall(r"_(\d{4}-\d{2}-\d{2})_", file)
                 if not date_match:
                     return None
                 return date_match[0]
             except Exception as e:
-                print(f"Error in get_date: {e}")
-                return None
+                msg = f"Error in get_date: {e}"
+                raise KiaraProcessingException(msg)
 
+        
         try:
-            sources = sources.with_columns([
-                sources[column_name].apply(get_date).alias('date'),
-                sources[column_name].apply(get_ref).alias('publication_ref')
-            ])
-        except Exception as e:
+            
+            pub_refs: list[str] = inputs.get_value_obj("map").data[0]
+            pub_names: list[str] = inputs.get_value_obj("map").data[1]
 
-            print(f"An error occurred while augmenting the dataframe: {e}")
+            pub_ref_to_name = dict(zip(pub_refs, pub_names))
+            
+            augm_sources = sources_tb.with_columns(
+                    sources_tb["publication_ref"]
+                    .map_elements(lambda x: pub_ref_to_name.get(x))
+                    .alias("publication_name")
+                )
 
-        try:
-            if pub_refs and pub_names:
-                pub_ref_to_name = dict(zip(pub_refs, pub_names))
-                sources = sources.with_columns(
-                    sources['publication_ref'].apply(lambda x: pub_ref_to_name.get(x, None)).alias('publication_name')
+        except:
```

Comment:
I must admit I don't 100% understand what is happening here, but I'm fairly sure that's just my brain not running properly now.

So, basically you try to add the human readable 'publication_name' column, and if that doesn't work you add the 'date' and 'pub_ref' columns? I don't really understand why you wouldn't always add those last two, but that's probably just me not understanding the scope within this module works. So you may just ignore this comment if it doesn't make sense.